### ISSUE TITLE
Be able to absorb 100s of new users at once

### ISSUE THREAD

**Issue by lourot** [AurelienLourot/ghuser.io#143]
Demand is high: https://news.ycombinator.com/item?id=17951478

**Comment by sam0x17** [comment]
as per my [comment on HN](https://news.ycombinator.com/item?id=17955373), a good way to do this might be using headless chrome via puppeteer from within a Google Cloud Function or using AWS Lambda. IP rate limiting might not be an issue given how the network of worker nodes is set up.

**Comment by sam0x17** [comment]
Yet another option might be to use Google's networking module to purchase a large pool of IPs and proxy through them via a cloud function as described above. I'm pretty sure that is possible.

**Comment by styfle** [comment]
You might want to reach out to @zeit because they have a system that can build and deploy any `Dockerfile` and then [auto scale](https://zeit.co/docs/getting-started/scaling#auto-scaling) as traffic increases. They will typically provide free or subsidized hosting for open source projects.

I am using zeit now to host a [somewhat popular project](https://github.com/styfle/packagephobia) with great success 😄 

https://zeit.co/now

**Comment by joecohens** [comment]
Maybe this can repo can help https://github.com/zeit/now-examples it's filled with lot of examples using `Dockerfile` deployments, also you can read more about it here https://zeit.co/blog/serverless-docker if you have questions let me know 😄

**Comment by olstenlarck** [comment]
Yup. Now is the way to go. Or at least if the current plans are don't not enough, contact them, they may support OSS with some customized plan.

Now is just freaking zero pain and zero efforts, 3 characters on a terminal and nothing more, and gives so freaking much in return. It's not just a hype or whatever. This company is really pawning and rocketing the things.

**Comment by ocdtrekkie** [comment]
One thing I would've recommended is to have a way to be notified when the service is available again when you removed the ability to join the queue. "We'll email you when your profile is ready" or even "we'll email you when our service can handle more profiles" would've been a better approach, though I am sure you were in a hurry to shut it off.

**Comment by AurelienLourot** [comment]
@ocdtrekkie Good idea, we have now created a twitter account: https://twitter.com/ghuserio . Thanks!

**Comment by michaelmior** [comment]
@AurelienLourot In case you weren't aware, Twitter is telling me that the profile is temporarily restricted when I try to visit.

**Comment by brillout** [comment]
We know and I just contacted Twitter about it https://twitter.com/brillout/status/1040643164774834176

**Comment by sam0x17** [comment]
Would love to use this if it is ready

**Comment by AurelienLourot** [comment]
@sam0x17 actually your profile is already there 😉 https://ghuser.io/sam0x17

@DiegoRBaquero yes still heavily working on scaling up. When this issue got opened we were serving about 200 profiles. Now there are 700 profiles and you are probably one of the 2000+ users who clicked "Get your profile" at some point, and we won't forget you 🙂 We're just kind of slow I admit. More details:

* [the scaling issues](https://github.com/ghuser-io/db/milestones/scaling%20issues) I'm working on as we speak
* [the roadmap](https://github.com/ghuser-io/ghuser.io#roadmap)
* the number of profiles is now visible on the [landing page](https://ghuser.io)

**Comment by aisensiy** [comment]
Waiting for my profile 😃

**Comment by reeversedev** [comment]
How soon can I expect that I'll be able to create my profile in ghuser.io? Would love to contribute as well.

**Comment by AurelienLourot** [comment]
:tada: we have just managed to catch up. Everyone who has ever clicked on 'Get your profile' up to yesterday should now have a profile. A few more facts:

* We are not overloaded anymore. We serve 3k users and with the current design we can grow up to 7k users. Probably easily 20k users with a few tweaks. The bottleneck is GitHub's API rate limit but we're not yet out of ideas to go beyond, e.g. see https://github.com/ghuser-io/db/issues/13
* About 10 people are joining every day at the moment. We don't have a satisfying automation implemented yet so when someone clicks on 'Get your profile', their username ends up in a queue which we process semi-automatically within 48 hours. Obviously this is one of the most urgent things to properly implement. This is what this issue is now about.
* The code base got dirtier at many places because we tried to rush at some point and we are now investing some time into cleaning this up.
* And as you may have noticed: we haven't stopped implementing new features :)

**Comment by brillout** [comment]
Update: Onboarding is currently stalled again: We are having an issue with our script that crawls GitHub (see https://github.com/ghuser-io/ghuser.io/issues/172). Good news though: We already have found a solution.

**Comment by AurelienLourot** [comment]
Update: #172 is fixed :tada: Onboarding is reopened.

**Comment by saddington** [comment]
cool. i shall wait patiently... 📦

**Comment by noxsphere** [comment]
no problem it was already a fake account

Le ven. 7 déc. 2018 à 01:12, John Saddington <notifications@github.com> a
écrit :
